odl/operator/default_ops.py

```diff
@@ -773,15 +777,23 @@ def __init__(self, vector, domain=None):
         >>> op(x, out=r3.element())
         rn(3).element([1.0, 2.0, 3.0])
         """
-        if not isinstance(vector, LinearSpaceElement):
-            raise TypeError('`vector` {!r} not a LinearSpaceElement instance'
-                            ''.format(vector))
+
+        if ((domain is None or range is None) and
+                not isinstance(vector, LinearSpaceElement)):
+                raise TypeError('If either domain or range is unspecified '
+                                '`vector` must be LinearSpaceVector, got {!r}.'
+                                ''.format(vector))
 
         if domain is None:
             domain = vector.space
+        if range is None:
+            range = vector.space
+        if vector not in range:
+            raise TypeError('`vector` {!r} not in range {}'
+                            ''.format(vector, range))
 
-        self.__vector = vector
-        super().__init__(domain, vector.space, linear=False)
+        super().__init__(domain, range)
+        self.__vector = self.__vector = self.range.element(vector)
```

Comment:
Plus we already check `vector in range` a few lines up, calling ´range.element` after that does nothing. If you want, you could remove the `vector in range` check above and defer it to simply doing a `vector = self.range.element(vector)` call.